index.d.ts

```diff
@@ -0,0 +1,209 @@
+    protocol: string
+    size: number
+    timeout: number
+  }
+
+  export interface RequestInit {
+    // whatwg/fetch standard options
+    body?: BodyInit
+    headers?: HeadersInit
+    method?: string
+    redirect?: RequestRedirect
+    signal?: AbortSignal | null
+
+    // node-fetch extensions
+    agent?: Agent | ((parsedUrl: URL) => Agent) // =null http.Agent instance, allows custom proxy, certificate etc.
+    compress?: boolean // =true support gzip/deflate content encoding. false to disable
+    follow?: number // =20 maximum redirect count. 0 to not follow redirect
+    size?: number // =0 maximum response body size in bytes. 0 to disable
+    timeout?: number // =0 req/res timeout in ms, it resets on redirect. 0 to disable (OS limit applies)
+
+    // node-fetch does not support mode, cache or credentials options
+  }
+
+  export type RequestContext =
+    | 'audio'
+    | 'beacon'
+    | 'cspreport'
+    | 'download'
+    | 'embed'
+    | 'eventsource'
+    | 'favicon'
+    | 'fetch'
+    | 'font'
+    | 'form'
+    | 'frame'
+    | 'hyperlink'
+    | 'iframe'
+    | 'image'
+    | 'imageset'
+    | 'import'
+    | 'internal'
+    | 'location'
+    | 'manifest'
+    | 'object'
+    | 'ping'
+    | 'plugin'
+    | 'prefetch'
+    | 'script'
+    | 'serviceworker'
+    | 'sharedworker'
+    | 'style'
+    | 'subresource'
+    | 'track'
+    | 'video'
+    | 'worker'
+    | 'xmlhttprequest'
+    | 'xslt'
+  export type RequestMode = 'cors' | 'no-cors' | 'same-origin'
+  export type RequestRedirect = 'error' | 'follow' | 'manual'
+  export type RequestCredentials = 'omit' | 'include' | 'same-origin'
+
+  export type RequestCache = 'default' | 'force-cache' | 'no-cache' | 'no-store' | 'only-if-cached' | 'reload'
+
+  export class Headers implements Iterable<[string, string]> {
+    constructor(init?: HeadersInit)
+    forEach(callback: (value: string, name: string) => void): void
+    append(name: string, value: string): void
+    delete(name: string): void
+    get(name: string): string | null
+    getAll(name: string): string[]
+    has(name: string): boolean
+    raw(): {[k: string]: string[]}
+    set(name: string, value: string): void
+
+    // Iterator methods
+    entries(): Iterator<[string, string]>
+    keys(): Iterator<string>
+    values(): Iterator<[string]>
+    [Symbol.iterator](): Iterator<[string, string]>
+  }
+
+  type BlobPart = ArrayBuffer | ArrayBufferView | Blob | string
+
+  interface BlobOptions {
+    type?: string
+    endings?: 'transparent' | 'native'
+  }
+
+  export class Blob {
+    constructor(blobParts?: BlobPart[], options?: BlobOptions)
+    readonly type: string
+    readonly size: number
+    slice(start?: number, end?: number): Blob
+  }
+
+  export class Body {
+    constructor(body?: any, opts?: {size?: number; timeout?: number})
+    arrayBuffer(): Promise<ArrayBuffer>
+    blob(): Promise<Buffer>
+    body: NodeJS.ReadableStream
+    bodyUsed: boolean
+    buffer(): Promise<Buffer>
```

Comment:
You should verify, but I don't think this library uses `Buffer` or `NodeJS.ReadableStream`.  I suspect this is an artifact of copying from `node-fetch` definition.